src/content/concepts/module-federation.md

```diff
@@ -0,0 +1,95 @@
+
+A container is created through a container entry, which exposes async access to specified modules. The exposed access is separated into two steps:
+
+1. loading the module (async)
+2. evaluating the module (sync).
+
+Step 1 will be done during chunk loading. Step 2 will be done during module evaluation interleaved with other (local and remote) modules. This way evaluation order is unaffected by converting a module from local to remote or the other way around.
+
+Nesting container is possible. Containers can use modules from other containers. Circular dependencies between container are also possible.
+
+### Overriding
+
+A container is able to flag selected local modules as "overridable". A consumer of the container is able to provide "overrides", which are modules that replace one of the overridable module of the container. All modules of the container will use the replacement module instead of the local module when the consumer provides one. When the consumer doesn't provide a replacement module, all modules of the container will use the local one.
+
+The container will organize overridable modules in a way that they do not need to be downloaded when they has been overriding by the consumer. This usually happens by placing them into separate chunks.
+
+One the other hand the provider of the replacement modules will only provide async loading functions which allow the container to load replacement modules only when they are needed. The provider will organize replacement modules in a way that they do not need to be downloaded when they are not requested by the container at all. This usually happens by placing them into separate chunks.
+
+A "name" is used to identify overridable modules from the container.
+
+Overrides are provided in a similar way as the container exposes modules, separated into two steps:
+
+1. Loading (async)
+2. evaluating (sync).
+
+In nested scenarios, where a container uses other containers to load modules from, is overriding a transitive operation. Providing overrides to one container will also override these modules in nested container. This means that using a container will automatically make all "names" of the overridable modules, part of the overriding interface of the parent container.
+
+Overrides must be provided before modules of the container are loaded, otherwise behavior is undefined.
+
+## High-level concepts
+
+Each build acts as container and also consumes other build as containers. This way each build is able to access any other exposed module by loading it from its container.
+
+Shared modules are modules that are both overridable and provided as overrides to nested container. They usually point to the same module in each build, e.g. the same library.
+
+## Building blocks
+
+### `OverridablesPlugin` (low level)
+
+This plugin makes specified modules "overridable". A local API (`__webpack_override__`) allows to provide overrides.
```

Comment:
This needs an example, we can skip it in this PR but nice to have.

Reply:
I've added one.